core/src/test/java/GameObjects/UrlBuilderImpTest.java

```diff
@@ -0,0 +1,95 @@
+package GameObjects;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the UrlBuilder. The URLs provided were taken directly from the database using their
+ * API.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UrlBuilderImpTest {
```

Comment:
I think the "foo" example would just use "foo" as the value.  For empty or null strings, I think urlBuilder should either return a null or empty string for the entire method to denote invalid settings, or throw an IllegalArgumentException.